src/libraries/System.Net.Sockets/tests/FunctionalTests/TcpListenerTest.cs

```diff
@@ -43,6 +43,28 @@ public void Active_TrueWhileRunning(int ctor)
             Assert.False(listener.Active);
         }
 
+        [Fact]
+        public void IDisposable_DisposeWorksAsStop()
+        {
+            var listener = new DerivedTcpListener(IPAddress.Loopback, 0);
+            using (listener)
+            {
+                Assert.False(listener.Active);
+                listener.Start();
+                Assert.True(listener.Active);
+            }
+            Assert.False(listener.Active);
+
+            listener = new DerivedTcpListener(IPAddress.Loopback, 0);
+            using ((IDisposable)listener)
+            {
+                Assert.False(listener.Active);
+                listener.Start();
+                Assert.True(listener.Active);
+            }
+            Assert.False(listener.Active);
```

Comment:
> Are we testing the same thing twice here?

`Dispose()` and `IDisposable.Dispose()` are sometimes different methods.

> Maybe you wanted the non-derived TcpListener in one of the cases?

Non-derived `TcpListener` does not have a public `Active` property so we can't check that the connection is closed.

Reply:
But `IDisposable.Dispose()` doesn't exist for this type (and I don't see why would we implement it in the future), so I don't see the value of this extra code in this particular case.

Reply:
Because tests should not rely on the current implementation. The public interface says nothing about how `IDisposable.Dispose()` is implemented now or in the future. It just also protects from incorrect implementation in the future.

Reply:
> Because tests should not rely on the current implementation. 

The presence or absence of a separate `IDisposable.Dispose()` is not an "implementation detail", it's an additional method. If anyone would want to add it in the future, they would need to first propose it, go through an API review, and implement it in a separate PR where new tests should be added for the new method.

There is no practice in the .NET repo to defensively test hypothetical implicit overloads, and I don't see why we would need to deviate from that in this case, especially that it's almost certain that we will never add an additional `IDisposable.Dispose()`.